iocore/net/SSLClientUtils.cc

```diff
@@ -37,27 +37,38 @@ typedef SSL_METHOD *ink_ssl_method_t;
 #endif
 
 int
-verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
+verify_callback(int signature_ok, X509_STORE_CTX *ctx)
 {
   X509 *cert;
   int depth;
   int err;
   SSL *ssl;
 
   SSLDebug("Entered verify cb");
-  depth = X509_STORE_CTX_get_error_depth(ctx);
-  cert  = X509_STORE_CTX_get_current_cert(ctx);
-  err   = X509_STORE_CTX_get_error(ctx);
 
   /*
    * Retrieve the pointer to the SSL of the connection currently treated
    * and the application specific data stored into the SSL object.
    */
   ssl                      = static_cast<SSL *>(X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx()));
   SSLNetVConnection *netvc = SSLNetVCAccess(ssl);
-  bool enforce_mode        = (netvc && netvc->options.clientVerificationFlag == static_cast<uint8_t>(YamlSNIConfig::Level::STRICT));
-  if (!preverify_ok && netvc != nullptr) {
-    // Don't bother to check the hostname if we failed openssl's verification
+
+  // No enforcing, go away
+  if (netvc && netvc->options.verifyServerPolicy == YamlSNIConfig::Policy::DISABLED) {
+    return true;       // Tell them that all is well
+  } else if (!netvc) { // No netvc, very bad.  Go away.  Things are not good.
+    return false;
+  }
+
+  depth = X509_STORE_CTX_get_error_depth(ctx);
+  cert  = X509_STORE_CTX_get_current_cert(ctx);
+  err   = X509_STORE_CTX_get_error(ctx);
+
+  bool enforce_mode = (netvc && netvc->options.verifyServerPolicy == YamlSNIConfig::Policy::ENFORCED);
```

Comment:
After line 59 no more checks of netvc being non-null are needed.

Reply:
True.  Will fix in other PR